### PR TITLE
Add DeBERTa head models

### LINKED ISSUE #9689: MLM training for DeBERTa not supported: configuration class is missing
When I ran the example script run_mlm.py to fine tune the pretrained deberta model on a customized dataset, I got the following error. The same command worked for roberta-base.

The command:
python run_mlm.py --model_name_or_path 'microsoft/deberta-base' --train_file slogans/train.txt --validation_file slogans/test.txt --do_train --do_eval --per_device_train_batch_size 64 --per_device_eval_batch_size 64 --learning_rate 1e-3 --num_train_epochs 10 --output_dir /home/jovyan/share2/xiaolin/models/mlm/temp --save_steps 5000 --logging_steps 100

The terminal error:
Traceback (most recent call last):
  File "run_mlm.py", line 409, in <module>
    main()
  File "run_mlm.py", line 264, in main
    cache_dir=model_args.cache_dir,
  File "/home/jovyan/.local/lib/python3.7/site-packages/transformers/models/auto/modeling_auto.py", line 1093, in from_pretrained
    config.__class__, cls.__name__, ", ".join(c.__name__ for c in MODEL_FOR_MASKED_LM_MAPPING.keys())
ValueError: Unrecognized configuration class <class 'transformers.models.deberta.configuration_deberta.DebertaConfig'> for this kind of AutoModel: AutoModelForMaskedLM.
Model type should be one of LayoutLMConfig, DistilBertConfig, AlbertConfig, BartConfig, CamembertConfig, XLMRobertaConfig, LongformerConfig, RobertaConfig, SqueezeBertConfig, BertConfig, MobileBertConfig, FlaubertConfig, XLMConfig, ElectraConfig, ReformerConfig, FunnelConfig, MPNetConfig, TapasConfig.

### PR DESCRIPTION
**Pull request by NielsRogge**
This PR adds 3 head models on top of the DeBERTa base model: `DebertaForMaskedLM`, `DebertaForTokenClassification`, `DebertaForQuestionAnswering`. These are mostly copied from `modeling_bert.py` with bert->deberta.

## Who can review?

@LysandreJik 

Also tagging original DeBERTa author: @BigBird01 

Fixes #9689

### CODE REVIEW

File: tests/test_modeling_deberta.py [hunk 1]
```diff
@@ -1,251 +1,290 @@
...
+from transformers.testing_utils import require_sentencepiece, require_tokenizers, require_torch, slow, torch_device
+
+from .test_configuration_common import ConfigTester
+from .test_modeling_common import ModelTesterMixin, ids_tensor
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import (  # XxxForMaskedLM,; XxxForQuestionAnswering,; XxxForTokenClassification,
```
Comment (LysandreJik) [inline tests/test_modeling_deberta.py:32]:
```suggestion
    from transformers import (
```

File: src/transformers/models/deberta/modeling_deberta.py [hunk 2]
```diff
@@ -1049,3 +1184,192 @@ def forward(
...
+                active_logits = logits.view(-1, self.num_labels)
+                active_labels = torch.where(
+                    active_loss, labels.view(-1), torch.tensor(loss_fct.ignore_index).type_as(labels)
+                )
+                loss = loss_fct(active_logits, active_labels)
+            else:
+                loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
+
+        if not return_dict:
+            output = (logits,) + outputs[2:]
```
Comment (LysandreJik) [inline src/transformers/models/deberta/modeling_deberta.py:1267]: If I'm not mistaken you copied this from the BERT modeling file, which has a pooler! The DeBERTa model doesn't have a pooler as the output of the base model suggests:

```
        return BaseModelOutput(
            last_hidden_state=sequence_output,
            hidden_states=encoder_outputs.hidden_states if output_hidden_states else None,
            attentions=encoder_outputs.attentions,
        )
```

It's weird that the hidden states tests didn't catch this ...

```suggestion
            output = (logits,) + outputs[1:]
```

File: src/transformers/models/deberta/modeling_deberta.py [hunk 3]
```diff
@@ -1049,3 +1184,192 @@ def forward(
...
+            start_positions.clamp_(0, ignored_index)
+            end_positions.clamp_(0, ignored_index)
+
+            loss_fct = CrossEntropyLoss(ignore_index=ignored_index)
+            start_loss = loss_fct(start_logits, start_positions)
+            end_loss = loss_fct(end_logits, end_positions)
+            total_loss = (start_loss + end_loss) / 2
+
+        if not return_dict:
+            output = (start_logits, end_logits) + outputs[2:]
```
Comment (LysandreJik) [inline src/transformers/models/deberta/modeling_deberta.py:1366]: Same here, I think this was copied from BERT!
```suggestion
            output = (start_logits, end_logits) + outputs[1:]
```

File: src/transformers/models/deberta/modeling_deberta.py [hunk 4]
```diff
@@ -945,6 +951,135 @@ def forward(
...
+        sequence_output = outputs[0]
+        prediction_scores = self.cls(sequence_output)
+
+        masked_lm_loss = None
+        if labels is not None:
+            loss_fct = CrossEntropyLoss()  # -100 index = padding token
+            masked_lm_loss = loss_fct(prediction_scores.view(-1, self.config.vocab_size), labels.view(-1))
+
+        if not return_dict:
+            output = (prediction_scores,) + outputs[2:]
```
Comment (LysandreJik) [inline src/transformers/models/deberta/modeling_deberta.py:1022]:
```suggestion
            output = (prediction_scores,) + outputs[1:]
```
(see comment below)

**Review by LysandreJik — COMMENTED** [review]
Fantastic work! Thanks for taking this up @NielsRogge, the implementation is very clean. You seem to have an issue with the `test_model_outputs_equivalence` test. Do you mind taking a look at it?

**Comment by NielsRogge** [comment]
Thanks for the review @LysandreJik, the test did fail because of the pooler. Is fixed now!

**Review by LysandreJik — APPROVED** [review]
Fantastic! Thanks for fixing. LGTM, merging!

**Pull request merged**